src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -15,125 +15,6651 @@
+    NPC_SUBDUED_FOREST_ETTIN                = 44737,
+    NPC_FORSAKEN_TROOPER_F                  = 44732,
+    NPC_FORSAKEN_TROOPER_M                  = 44733,
+
+    SPELL_CHAIN_RIGHT                       = 83467,
+    SPELL_CHAIN_LEFT                        = 83464,
+    SPELL_EJECT_PASSENGERS_3_8              = 83477,
 
-    SAY_RANE            = 0,
-    SAY_RANE_ANSWER     = 5,
-    SAY_MOVE_QUINN      = 6,
+    EVENT_TO_FORSAKEN_HIGH_COMMAND          = 1,
+    EVENT_TO_THE_SEPULCHER                  = 2,
+    EVENT_TO_THE_FORSAKEN_FRONT             = 3,
+    EVENT_TO_DESPAWN                        = 4,
+    EVENT_YELL_1                            = 5,
+    EVENT_YELL_2                            = 6,
+    EVENT_YELL_3                            = 7,
 
-    SAY_QUINN           = 7,
-    SAY_QUINN_ANSWER    = 0,
-    SAY_BYE             = 8,
+    DATA_WAYPOINT_TO_FORSAKEN_HIGH          = 447310,
+    DATA_WAYPOINT_TO_SEPULCHER              = 447311,
+    DATA_WAYPOINT_TO_FORSAKEN_FRONT         = 447312,
+    DATA_WAYPOINT_TO_DESPAWN                = 447313,
 
-    QUEST_ESCORTING     = 435,
-    NPC_RANE            = 1950,
-    NPC_QUINN           = 1951
+    DATA_ANIMKIT_HORDE_ENGINEER             = 91298
 };
 
-class npc_deathstalker_erland : public CreatureScript
+// Horde Hauler - 44731
+struct npc_silverpine_horde_hauler : public ScriptedAI
 {
-public:
-    npc_deathstalker_erland() : CreatureScript("npc_deathstalker_erland") { }
+    npc_silverpine_horde_hauler(Creature* creature) : ScriptedAI(creature)
+    {
+        Initilize();
+    }
+
+    void Initilize()
+    {
+        if (Creature* engineer = me->FindNearestCreature(NPC_HORDE_ENGINEER, 15.0f))
+        {
+            if (Creature* ettin = me->FindNearestCreature(NPC_SUBDUED_FOREST_ETTIN, 15.0f))
+            {
+                engineer->CastSpell(ettin, SPELL_CHAIN_RIGHT, true);
+                engineer->CastSpell(ettin, SPELL_CHAIN_LEFT, true);
+                engineer->AddAura(DATA_ANIMKIT_HORDE_ENGINEER, engineer);
+
+                _events.ScheduleEvent(EVENT_TO_FORSAKEN_HIGH_COMMAND, 5s);
+            }
+        }
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
 
-    struct npc_deathstalker_erlandAI : public EscortAI
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
     {
-        npc_deathstalker_erlandAI(Creature* creature) : EscortAI(creature) { }
+        if (!apply)
+            return;
 
-        void WaypointReached(uint32 waypointId, uint32 /*pathId*/) override
+        _playerGUID = passenger->GetGUID();
+
+        if (seatId == 2)
         {
-            Player* player = GetPlayerForEscort();
-            if (!player)
-                return;
+            if (Creature* engineer = me->FindNearestCreature(NPC_HORDE_ENGINEER, 15.0f, true))
+            {
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    engineer->AI()->Talk(TALK_0, player);
```

Comment:
missing check if AI is enabled. Can crash the server if the engineer is mind controlled in the right moment.